unlock-app/src/components/interface/locks/Settings/forms/KeyGranterForm.tsx

```diff
@@ -0,0 +1,197 @@
+import { useQuery } from '@tanstack/react-query'
+import {
+  Button,
+  AddressInput,
+  isAddressOrEns,
+  Placeholder,
+} from '@unlock-protocol/ui'
+import { SubgraphService } from '@unlock-protocol/unlock-js'
+import { Controller, useForm, useWatch } from 'react-hook-form'
+import { BsCreditCard as CreditCardIcon } from 'react-icons/bs'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { onResolveName } from '~/utils/resolvers'
+import useEns from '~/hooks/useEns'
+import { useAddKeyGranter, useRemoveKeyGranter } from '~/hooks/useKeyGranter'
+
+interface KeyGranterFormProps {
+  lockAddress: string
+  network: number
+  isManager: boolean
+  disabled: boolean
+}
+
+interface KeyGranterCardProps {
+  lockAddress: string
+  network: number
+  keyGranter: string
+  isManager: boolean
+}
+
+const KeyGranterCard = ({
+  lockAddress,
+  network,
+  keyGranter,
+  isManager,
+}: KeyGranterCardProps) => {
+  const { account } = useAuth()
+
+  const keyGranterEnsOrAddress = useEns(keyGranter)
+  const isLoggedUser = account?.toLowerCase() === keyGranter?.toLowerCase()
+  const PURCHASE_ADDRESSES = [
+    '0x22c095c69c38b66afAad4eFd4280D94Ec9D12f4C', // prod purchaser
+    '0x903073735Bb6FDB802bd3CDD3b3a2b00C36Bc2A9', // staging purchaser
+    '0xd851fe9ba8EfA66e65d7865690bD2B9522C6E99f', // OpenZeppelin purchaser
+  ].map((add) => add.toLowerCase())
+  const isCreditCardKeyGranter = PURCHASE_ADDRESSES.includes(keyGranter)
+
+  const removeKeyGranterMutation = useRemoveKeyGranter(lockAddress, network)
+
+  const removeKeyGranter = async () => {
+    await removeKeyGranterMutation.mutateAsync(keyGranter)
+  }
+
+  return (
+    <div className="flex items-center justify-between px-4 py-2 border border-gray-200 rounded-lg">
+      <div className="flex flex-col gap-2 ">
+        <span className="text-base text-brand-dark">
+          {keyGranterEnsOrAddress}{' '}
+          {isCreditCardKeyGranter ? <CreditCardIcon size={18} /> : null}
```

Comment:
In the case where this is is the lockmsnith's key granter, let's not even show its address but just "Credit card granter", with the icon!